### PR TITLE
[use case] User gets revoked

### LINKED ISSUE #184: [use case] user gets revoked from a repo
- [ ] user has no projects on its scope.
- [ ] user interacts with a project which repo isn't in its scope.
- [ ] anything else somebody may propose

### PR DESCRIPTION
**Pull request by minuscorp**
Fixes #184

### CODE REVIEW

File: docs/use_cases/user_gets_revoked.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# User gets revoked from a project #
+
+This happens whenever an user gets its permissions over the repository removed. This use case has a side case in where an user might not be able to see any project in the dashboard due to lack of permissions to any project.
```
Comment (KrauseFx) [inline docs/use_cases/user_gets_revoked.md]: `removed` -> `revoked` I think?

File: docs/use_cases/user_gets_revoked.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+## Assumptions ##
+
+- User logged in.
+- Dashboard might show a particular list of 1 or more projects which the user has access to.
+- **OR**
+- Dashboard might show an user-friendly UI where it gets informed that he has no access to any projects on its scope: _"If you think you should be seeing projects here, contact with your organization administrator and request the appropriate permissions."_, or the start point of **creating a new project** use case.
```
Comment (KrauseFx) [inline docs/use_cases/user_gets_revoked.md]: `contact your GitHub administrator`

File: docs/use_cases/user_gets_revoked.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# User gets revoked from a project #
+
+This happens whenever an user gets its permissions over the repository removed. This use case has a side case in where an user might not be able to see any project in the dashboard due to lack of permissions to any project.
+
+## Assumptions ##
+
+- User logged in.
+- Dashboard might show a particular list of 1 or more projects which the user has access to.
```
Comment (taquitos) [inline docs/use_cases/user_gets_revoked.md]: I think the only assumptions are:
-  User logged in.
-  User at dashboard

File: docs/use_cases/user_gets_revoked.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+- **OR**
+- Dashboard might show an user-friendly UI where it gets informed that he has no access to any projects on its scope: _"If you think you should be seeing projects here, contact with your organization administrator and request the appropriate permissions."_, or the start point of **creating a new project** use case.
+
+## Side-effects ##
+
+1. _Dashboard_ checks for permissions on reload on every project with the user credentials.
+2. If the user tries to access to a project which he has no longer permissions over, it should receive a nice _403 Forbidden_ message about he has no access to the selected project.
```
Comment (KrauseFx) [inline docs/use_cases/user_gets_revoked.md]: he -> they
and then `he has` -> `they have`

Comment (KrauseFx) [reply docs/use_cases/user_gets_revoked.md]: Gender neutral language :)

File: docs/use_cases/user_gets_revoked.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+## Assumptions ##
+
+- User logged in.
+- Dashboard might show a particular list of 1 or more projects which the user has access to.
+- **OR**
+- Dashboard might show an user-friendly UI where it gets informed that he has no access to any projects on its scope: _"If you think you should be seeing projects here, contact with your organization administrator and request the appropriate permissions."_, or the start point of **creating a new project** use case.
+
+## Side-effects ##
+
+1. _Dashboard_ checks for permissions on reload on every project with the user credentials.
```
Comment (KrauseFx) [inline docs/use_cases/user_gets_revoked.md]: So for now we want to check the permissions for all projects from `projects.json` every time the dashboard is opened?

File: docs/use_cases/user_gets_revoked.md (hunk 3 above)
Comment (taquitos) [inline docs/use_cases/user_gets_revoked.md]: We can move this to a `### Steps ###` section like:
1. If user has no create access, user sees message `...`
1. If user has create access, but no other repo access, user sees message `...`

File: docs/use_cases/user_gets_revoked.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+This happens whenever an user gets its permissions over the repository removed. This use case has a side case in where an user might not be able to see any project in the dashboard due to lack of permissions to any project.
+
+## Assumptions ##
+
+- User logged in.
+- Dashboard might show a particular list of 1 or more projects which the user has access to.
+- **OR**
+- Dashboard might show an user-friendly UI where it gets informed that he has no access to any projects on its scope: _"If you think you should be seeing projects here, contact with your organization administrator and request the appropriate permissions."_, or the start point of **creating a new project** use case.
+
+## Side-effects ##
```
Comment (taquitos) [inline docs/use_cases/user_gets_revoked.md]: `## Side-effects ##` are more of a technical specification thing, use cases are really just `how will a user expect to interact with the system?`

File: docs/use_cases/user_gets_revoked.md (hunk 4 above)
Comment (taquitos) [inline docs/use_cases/user_gets_revoked.md]: This is great, I'd put it in the `Steps` section as:
`1.  Or, if user is at a project details link but has no access, they see <...>`

We don't want to have specific tech details like `403`. Think of use cases as more of a UX/UI thing. Designers don't need to know about 403s 😄

File: docs/use_cases/user_gets_revoked.md (hunk 5 above)
Comment (taquitos) [reply docs/use_cases/user_gets_revoked.md]: Technically speaking, I think we can lazily do that at the start. If you click on a link to a project, we can check the permission.

Comment (KrauseFx) [reply docs/use_cases/user_gets_revoked.md]: Yep, works! 👍

File: docs/use_cases/user_gets_revoked.md (hunk 4 above)
Comment (minuscorp) [reply docs/use_cases/user_gets_revoked.md]: Sorry, bad habits from Spanish 😅

File: docs/use_cases/user_gets_revoked.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+This happens whenever an user gets its permissions over the repository revoked. This use case has a side case in where an user might not be able to see any project in the dashboard due to lack of permissions to any project.
+
+## Assumptions ##
+
+- User logged in.
+- User at dashboard.
+
+## Steps ##
+
+1. If the user has no create permissions, and there're no projects, the dashboard might show an user-friendly UI where it gets informed that he has no access to any projects on its scope: _"If you think you should be seeing projects here, contact your GitHub administrator and request the appropriate permissions."_.
```
Comment (KrauseFx) [inline docs/use_cases/user_gets_revoked.md]: What's a `create permission`, maybe should be rephrased?

Comment (minuscorp) [reply docs/use_cases/user_gets_revoked.md]: I was thinking in something like permissions on a file system, `read|write|execution` permissions. Is that the correct way to write it?

Comment (taquitos) [reply docs/use_cases/user_gets_revoked.md]: Permissions here are 100% based on github. I think they have two: read/write
We shouldn't store any permission information though, we can just check github to see what permissions they have so we don't have to worry about enforcing permissions. We just have to check permissions to know what is the proper message is to display.

Comment (minuscorp) [reply docs/use_cases/user_gets_revoked.md]: So may I rephrase it like: 
> If the user has no write permissions over the organization, and...

Comment (KrauseFx) [inline docs/use_cases/user_gets_revoked.md]: `there're` spelling mistake

**Pull request merged**